compositor_pipeline/src/pipeline/encoder.rs

```diff
@@ -160,13 +157,22 @@ pub struct PacketIterator<'a> {
 }
 
 impl<'a> Iterator for PacketIterator<'a> {
-    type Item = Packet;
+    type Item = EncodedChunk;
 
     fn next(&mut self) -> Option<Self::Item> {
         let mut packet = Packet::empty();
 
         if self.encoder.encoder.receive_packet(&mut packet).is_ok() {
-            Some(packet)
+            match EncodedChunk::from_av_packet(&packet, EncodedChunkKind::Video(Codec::H264)) {
+                Ok(chunk) => Some(chunk),
+                Err(e) => {
+                    warn!(
+                        "failed to parse an ffmpeg packet received from encoder: {}",
+                        e
```

Comment:
```suggestion
                        "failed to parse an ffmpeg packet received from encoder: {e}"
```
formatting looks a bit wierd